common/src/main/java/org/conscrypt/OpenSSLXDHKeyFactory.java

```diff
@@ -60,6 +65,10 @@ protected PrivateKey engineGeneratePrivate(KeySpec keySpec) throws InvalidKeySpe
         if (keySpec instanceof PKCS8EncodedKeySpec) {
             return new OpenSSLX25519PrivateKey((PKCS8EncodedKeySpec) keySpec);
         }
+        if (keySpec instanceof SecretKeySpec
```

Comment:
The docs suggest this is more for symmetric keys. Or is this also used for asymmetric ones? The reference implementation's JEP made separate KeySpec types, which suggests SecretKeySpec isn't meant to be applied here: https://openjdk.org/jeps/324

@prbprbprb do you know the expectations here?